google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java

```diff
@@ -283,11 +282,12 @@ public void uploadFrom(Path path, BlobWriteOption... options) {
    * @param chunkSize the minimum size that will be written by a single RPC.
    * @param options blob write options
    * @throws StorageException upon failure
-   * @throws IllegalArgumentException if file does not exist
    */
   public void uploadFrom(Path path, int chunkSize, BlobWriteOption... options) {
+    if (!Files.exists(path)) {
+      throw new StorageException(0, "File to upload from does not exist '" + path + "'");
```

Comment:
you can add exception message like `The specified file does not exist on '" + path + "'");`instead of `"File to upload from does not exist '" + path + "'"`

Reply:
Thanks, I updated the wording to be close to NIO exceptions. I also added handling the case of uploading from a directory.